pkg/gpio/gpio_linux.go

```diff
@@ -39,10 +40,49 @@ func (v Value) String() string {
 	return "1"
 }
 
+// GetPinID is used to get the ID for a specific GPIO. This is done by
+// getting the base offset of the GPIO controller chip that covers this pin, and
+// adding the pin number.
+func GetPinID(controller string, pin int) (int, error) {
+	controllers, err := filepath.Glob(fmt.Sprintf("%s/gpiochip*", gpioPath))
+	if err != nil {
+		return 0, err
+	}
+
+	for _, c := range controllers {
+		// Get label (name of the controller)
+		buf, err := ioutil.ReadFile(filepath.Join(c, "label"))
+		if err != nil {
+			return 0, fmt.Errorf("failed to read label of %s: %v", c, err)
+		}
+		label := strings.TrimSpace(string(buf))
+
+		// Check that this is the controller we want
+		if strings.TrimSpace(label) != controller {
+			continue
+		}
+
+		// Get base offset (the first GPIO managed by this chip)
+		buf, err = ioutil.ReadFile(filepath.Join(c, "base"))
+		if err != nil {
+			return 0, fmt.Errorf("failed to read base of %s: %v", c, err)
+		}
+		baseStr := strings.TrimSpace(string(buf))
+		base, err := strconv.Atoi(baseStr)
+		if err != nil {
+			return 0, fmt.Errorf("could not get base offset for %s: %v", c, err)
+		}
+
+		return base + pin, nil
+	}
+
+	return 0, fmt.Errorf("could not find controller %s", controller)
+}
+
 // SetOutputValue configures the gpio as an output pin with the given value.
 func SetOutputValue(pin int, val Value) error {
 	dir := val.Dir()
-	path := filepath.Join(gpioPath, "gpio", strconv.Itoa(pin), "direction")
+	path := filepath.Join(gpioPath, fmt.Sprintf("gpio%d", pin), "direction")
```

Comment:
did this not already go in?

Reply:
It should have, I can see it in master, I'm not sure why that commit is being added here...